worker/dbaccessor/tracker_test.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package dbaccessor
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/worker/v3/workertest"
+	gc "gopkg.in/check.v1"
+
+	coredatabase "github.com/juju/juju/core/database"
+	"github.com/juju/juju/testing"
+)
+
+type trackedDBWorkerSuite struct {
+	dbBaseSuite
+}
+
+var _ = gc.Suite(&trackedDBWorkerSuite{})
```

Comment:
We need a test that we can set the poll interval and it will actually verify the database multiple times. I believe with your current setup it only ever pings the database 1 time, and that is 10s after it starts, but never again.